compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/IrInlineCodegen.kt

```diff
@@ -72,32 +73,24 @@ class IrInlineCodegen(
                 activeLambda = null
             }
         } else {
-            putValueOnStack(argumentExpression, parameterType, irValueParameter.index, blockInfo, irValueParameter.type)
+            // Reusing the same local reduces the amount of debug information available when breaking
+            // inside the inline function. Here we replicate the behavior of the non-IR backend, which at least
+            // preserves the receivers.
+            val onStack = if (irValueParameter.index >= 0)
```

Comment:
@pyos Could you clarify a bit this line and comment for it 'which at least preserves the receivers.': 
as i remember receivers have negative index so 'if (irValueParameter.index >= 0)' looks wrong

Reply:
...Or is it assumes that all receivers are materialized on stack?

Reply:
If so then 'if (irValueParameter.index >= 0)' check could be removed: it was impossible to make such optimization in old backend cause generally speaking both receivers (dispatch+extension) were incapsulated in one StackValue/or materialized before inline stuff

Reply:
The condition is correct: if index < 0 (=> receiver) we materialize the value (== call `gen`), otherwise we ask ExpressionCodegen to return an unmaterialized `StackValue.local` when possible. It's OK to always use `genOrGetLocal` here from the correctness perspective, yes, although it would cause the debugger to not show there receiver when stopped at a breakpoint in an inline method, and also change the output of some bytecode text tests.

Reply:
Than it's ok, more precise tuning and debugger support could be made late. Could you just add comment about additional optimization?

Reply:
Rewrote the comment so that it is hopefully more clear